diesel/src/pg/connection/raw.rs

```diff
@@ -167,3 +170,37 @@ impl Drop for PgString {
         }
     }
 }
+
+/// Internal wrapper around a `*mut PGresult` which is known to be not-null, and
+/// have no aliases.  This wrapper is to ensure that it's always properly
+/// dropped.
+///
+/// If `Unique` is ever stabilized, we should use it here.
+#[allow(missing_debug_implementations)]
+pub struct RawResult(*mut PGresult);
+
+unsafe impl Send for RawResult {}
+unsafe impl Sync for RawResult {}
+
+impl RawResult {
+    fn new(ptr: *mut PGresult, conn: &RawConnection) -> QueryResult<Self> {
```

Comment:
Reading the code above, it took me a while to figure out that `RawResult::new` returned a `QueryResult`, because it really surprised me to see a `new` method that returns a `Result`.

I don't think this is very idiomatic, but it's also what `std::ffi::CString` does, so I guess it's okay. I would probably call this method `try_new`, or `try_form` myself (sadly, `std::convert::TryFrom` is still unstable).